src/utilities/util_dump.c

```diff
@@ -805,15 +842,328 @@ dump_all_records(WT_CURSOR *cursor, bool reverse, bool json)
             if (fputc(',', fp) == EOF)
                 return (util_err(session, EIO, NULL));
         }
-        print_record(cursor, json);
+        WT_RET(print_record(cursor, json));
         once = true;
     }
 
+    if (ret != WT_NOTFOUND)
+        return (util_err(session, ret, reverse ? "WT_CURSOR.prev" : "WT_CURSOR.next"));
+
     if (json && once && fprintf(fp, "\n") < 0)
         return (util_err(session, EIO, NULL));
     return (0);
 }
 
+/*
+ * dump_explore --
+ *     Dump data in an interactive fashion.
+ */
+static int
+dump_explore(WT_CURSOR *cursor, const char *uri, bool reverse, bool pretty, bool hex, bool json)
+{
+    WT_DECL_RET;
+    WT_SESSION *session;
+    uint64_t bookmark_index, window;
+    int i, exact, num_args;
+    char *args[MAX_ARGS], *bookmarks[MAX_BOOKMARKS];
+    char *first_arg, user_input[MAX_ARGS], *current_arg;
```

Comment:
Possibly based on the naming should this be something like?:
```suggestion
    char *first_arg, user_input[ARG_BUF_SIZE], *current_arg;
```